### PR TITLE
:book: added CRD generation reference for v2 docs

### PR DESCRIPTION
**Pull request by droot**
This change copies the reference docs for CRD generation from v1
to v2 with minimal changes.

### CODE REVIEW

File: docs/book/src/reference/generating-crd.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+Note that `description`, `priority` and `format` are optional. Refer to the
+[additonal printer columns docs](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#additional-printer-columns)
+to learn more about the values of `name`, `type`, `JsonPath`, `description`, `priority` and `format`.
+
+The following example adds the `Spec`, `Replicas`, and `Age` columns.
+
+```Go
+// +kubebuilder:printcolumn:name="Spec",type="integer",JSONPath=".spec.cronSpec",description="status of the kind"
+// +kubebuilder:printcolumn:name="Replicas",type="integer",JSONPath=".spec.Replicas",description="The number of jobs launched by the CronJob"
```
Comment (mengqiy) [inline docs/book/src/reference/generating-crd.md]: > JSONPath=".spec.Replicas"

`Replicas` should be in lower case.

File: docs/book/src/reference/generating-crd.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+Format: `// +kubebuilder:printcolumn:name="Name",type="type",JSONPath="json-path",description="desc",priority="priority",format="format"`
+
+Note that `description`, `priority` and `format` are optional. Refer to the
+[additonal printer columns docs](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#additional-printer-columns)
+to learn more about the values of `name`, `type`, `JsonPath`, `description`, `priority` and `format`.
+
+The following example adds the `Spec`, `Replicas`, and `Age` columns.
+
+```Go
```
Comment (mengqiy) [inline docs/book/src/reference/generating-crd.md]: `s/Go/go` to be consistent with other places.

File: docs/book/src/reference/generating-crd.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+endif
+```
+
+When you run `make manifests`, you should see generated CRDs are under `config/crd/bases` directory.
+
+`controller-gen` generates manifests for RBAC as well, but this section covers the generation of CRD manifests.
+
+`controller-gen` reads kubebuilder markers of the form `// +kubebuilder:something...` defined as Go comments in the `<your-api-kind>_types.go` file under `apis/...` to produce the CRD manifests. Sections below describe various supported annotations.
+
+## Validation
```
Comment (mengqiy) [inline docs/book/src/reference/generating-crd.md]: We should mention the differences between pre-1.13 and 1.13+ and how to use `crd:trivialVersions` in controller-gen.

Comment (droot) [reply docs/book/src/reference/generating-crd.md]: Added a section for multiple version support.

**Pull request merged**